appinventor/appengine/war/login.jsp

```diff
@@ -1,15 +1,16 @@
 <%@page import="javax.servlet.http.HttpServletRequest"%>
 <%@page import="com.google.appinventor.server.util.UriBuilder"%>
+<%@page import="org.apache.commons.lang3.StringEscapeUtils"%>
 <%@page contentType="text/html" pageEncoding="UTF-8"%>
 <!doctype html>
 <%
-   String error = request.getParameter("error");
-   String useGoogleLabel = (String) request.getAttribute("useGoogleLabel");
-   String locale = request.getParameter("locale");
-   String redirect = request.getParameter("redirect");
-   String repo = (String) request.getAttribute("repo");
-   String autoload = (String) request.getAttribute("autoload");
-   String galleryId = (String) request.getAttribute("galleryId");
+   String error = StringEscapeUtils.escapeHtml4(request.getParameter("error"));
+   String useGoogleLabel = StringEscapeUtils.escapeHtml4((String) request.getAttribute("useGoogleLabel"));
```

Comment:
Given that `useGoogleLabel` is never output--its only used to determine whether we should show more content--I don't think it's necessary to escape it here. I also think it's only ever set on the server side by trusted code (as far as you trust our code). I'm happy to accept it as is though.